src/sentry/snuba/query_subscription_consumer.py

```diff
@@ -38,6 +59,151 @@ def inner(func: TQuerySubscriptionCallable) -> TQuerySubscriptionCallable:
     return inner
 
 
+def parse_message_value(value: str) -> Dict[str, Any]:
+    """
+    Parses the value received via the Kafka consumer and verifies that it
+    matches the expected schema.
+    :param value: A json formatted string
+    :return: A dict with the parsed message
+    """
+    with metrics.timer("snuba_query_subscriber.parse_message_value.json_parse"):
+        wrapper: Dict[str, Any] = json.loads(value)
+
+    with metrics.timer("snuba_query_subscriber.parse_message_value.json_validate_wrapper"):
+        try:
+            jsonschema.validate(wrapper, SUBSCRIPTION_WRAPPER_SCHEMA)
```

Comment:
Not for now as there's enough changes happening here but later on you could use https://github.com/getsentry/arroyo/blob/56066e373bd196482cd40ded8a5a613a7486a202/arroyo/processing/strategies/decoder/json.py in Arroyo for this. It will probably be faster for no other reason but that it uses fastjsonschema.